classes/core/AltapayReconciliation.php

```diff
@@ -152,29 +171,23 @@ public function exportReconciliationCSV() {
 					$year  = (int) substr( $yearMonth, 0, 4 );
 					$month = (int) substr( $yearMonth, 4, 2 );
 
-					$args['date_query'] = array(
-						array(
-							'year'  => $year,
-							'month' => $month,
-						),
-					);
+					$last_day_of_month    = date_create( "$year-$month" )->format( 'Y-m-t' );
+					$args['date_created'] = "$year-$month-01..." . $last_day_of_month;
 				}
 			}
 
-			$query = new \WP_Query( $args );
-
-			if ( $query->have_posts() ) {
-
-				$PostToSelect = substr( str_repeat( ',%d', count( $query->posts ) ), 1 );
+			$ordersData = wc_get_orders( $args );
 
-				$reconciliation_data =
-					$wpdb->get_results(
-						$wpdb->prepare(
-							"SELECT orderId, identifier, transactionType FROM {$wpdb->prefix}altapayReconciliationIdentifiers WHERE orderId IN ($PostToSelect) ",
-							$query->posts
-						),
-						ARRAY_A
-					);
+			if ( ! empty( $ordersData ) ) {
+				$orders              = $ordersData->orders;
+				$orders_to_select 	 = substr( str_repeat( ',%d', count( $orders ) ), 1 );
+				$reconciliation_data = $wpdb->get_results(
+					$wpdb->prepare(
+						"SELECT orderId, identifier, transactionType FROM {$wpdb->prefix}altapayReconciliationIdentifiers WHERE orderId IN ($orders_to_select) ",
+						$orders
+					),
```

Comment:
why are you concatenating manually here? Can't you pass them as parameters instead?

Example: $wpdb->prepare

Reply:
@emicha The $orders_to_select variable dynamically generates placeholders such as `%d,%d,%d,%d,%d` based on the number of elements in the $orders array. These placeholders are then used within the query, and the actual values 

So the prepare statement executes as:

```
$wpdb->prepare(
  "SELECT orderId, identifier, transactionType FROM wpap_altapayReconciliationIdentifiers WHERE orderId IN (%d,%d,%d,%d,%d)",
  array(11357,11355,11351,11349,11348)
)
```